docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+#### Inline UoM definition
+When the units of measure are specified directly in metadata's configuration file (i.e. inline definition), the actual units of measure are provided in core metadata's configuration (currently in `res/configuration.toml`) - see example B below.  
+
+**Example B Metadata Configuration - inline definition of UoM**
+``` toml
+[Writable]
+    [Writable.UoM]
+    validation=false  ## off by default
+
+## in the non-writable area - example inline units of measure
+[Uom]
+    source="reference to source for all UoM if not specified below"
+    [Uom.temperature]
+    source="www.weather.com"
+    values="C,F,K"
+    [Uom.weights]
+    source="www.usa.gov/federal-agencies/weights-and-measures-division"
+    values="lbs,ounces,kilos,grams"
+```
+
+!!! Note
+    When the units of measure are defined in-line, then *environmental overrides can be used to specify and override the units of measure in the configuration.toml file*.  However, since the Uom is a map, the unit of measure category must already exist in order to be overridden.  In the example above, you could provide an environmental override to replace `Uom.weights` but you could not provide an override for `Uom.speed` as it does not exist in the original configuration file.
+
+#### Specifying Validation on or off
+Additionally, in either external or inline definition, a configuration option for unit of measure validation being `on` or `off` will be provided (note `validation` in both example A and B above).  The units of measure or the location of the UoM file is static, but the ability to turn validation on/off is dynamic and therefore in the writable area of configuration.  For backward compatibility, validation will be off by default.
+
+Again, when a device resource does not have a `units` specified, this is still considered "valid".  So in reality, the diagram below specifies how validation is done by core metadata.
+
+![image](EdgeX_UoM_Validation.png)
+
+### Validation of the Units of Measure
+
+Core metadata will read the units of measure from its configuration (either inline or from the specified external file).  Like all configuration information, this data will be stored in the configuration service (Consul today) on initial startup of the core metadata service.
+
+When validation is turned `on` (Writable.UoM.validation is set to true), all device profile `units` (in device resource, device properties) will be validated against the list of units of measure by core metadata.  In other words, when a device profile is created or updated or when a device resource is added or updated via the core metadata API, the units specified in the device resource's `units` field (see resource example below) will be checked against the valid list of UoM provided via core metadata configuration.  If the `units` value matches any one of the configuration units of measure, then the device resource is considered valid - allowing the create or update operation to continue.  
```

Comment:
I think as part of our device profile changes ADR, I'm pretty sure we added `Units` to `Reading`? If so, should we also consider validation of readings?

Reply:
Discussed in both the 2/17/22 and 2/24/22 core working group meetings.  There is not a need to validate the readings as the device services will prohibit the creation of a reading without one of the specified units of measure.  So as long as validation is on and a unit of measure is specified in the device resource (it is optional - and if not provided then it is assumed you don't care what the uom is), no reading will get created with an invalid unit.